node/core/dispute-coordinator/src/lib.rs

```diff
@@ -210,31 +207,20 @@ impl DisputeCoordinatorSubsystem {
 		B: Backend + 'static,
 	{
 		loop {
-			let db_params =
-				DatabaseParams { db: self.store.clone(), db_column: self.config.col_session_data };
-
-			let (first_leaf, rolling_session_window) =
-				match get_rolling_session_window(ctx, db_params).await {
-					Ok(Some(update)) => update,
-					Ok(None) => {
-						gum::info!(target: LOG_TARGET, "received `Conclude` signal, exiting");
-						return Ok(None)
-					},
-					Err(e) => {
-						e.split()?.log();
-						continue
-					},
-				};
+			let first_leaf = match wait_for_first_leaf(ctx).await {
+				Ok(Some(activated_leaf)) => activated_leaf,
+				Ok(None) => continue,
+				Err(e) => {
+					e.split()?.log();
+					continue
+				},
+			};
 
+			let mut runtime_info = RuntimeInfo::new(None);
```

Comment:
Cache is 2 by default, we need it to be 6 - use `new_with_config` instead.

Reply:
worth considering pre-caching in dispute-distribution as well.

Reply:
This is a very good remark. I wrongly assumed that `RuntimeInfo` relies on the cache of `runtime-api-subsystem` (which caches results from runtime calls) and I blindly ignored its implementation :see_no_evil: 
But `RuntimeInfo` has got its own cache:
https://github.com/paritytech/polkadot/blob/27ddd270c3979fd2a5e2ac6fb3c9d474962c9260/node/subsystem-util/src/runtime/mod.rs#L169
which actually changes a lot of things (for the better).

Reply:
> worth considering pre-caching in dispute-distribution as well 

I've opened a separate PR for this: https://github.com/paritytech/polkadot/pull/6993